maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java

```diff
@@ -320,12 +320,14 @@ void initialize(CliRequest cliRequest) throws ExitException {
         // We need to locate the top level project which may be pointed at using
         // the -f/--file option.  However, the command line isn't parsed yet, so
         // we need to iterate through the args to find it and act upon it.
-        Path topDirectory = Paths.get(cliRequest.workingDirectory);
+        Path cwd = Paths.get(cliRequest.workingDirectory);
+        // MavenCliTest extensively uses MULTIMODULE_PROJECT_DIRECTORY
+        Path topDirectory = cliRequest.multiModuleProjectDirectory.toPath();
```

Comment:
That's wrong.  We can't have `topDirectory` depends on the deprecated `multiModuleProjectDirectory`.

Reply:
We have it solely due MavenCLITest, that excessively uses multiModuleProjectDirectory System Property. 

In "normal" uses it is NOT, as in " normal" case you'd have cmd line with -f etc, but to make UT pass unmodified, I had to make topDir == multiModule (as prop is set in UT).

Reply:
In addition, having everything depending on a system property which is set externally, and thus, out of control, is a bad idea.

Reply:
In other words, in "normal" case you'd have full CLI args and all and they would be same, they would end up same. It is ONLY the UT, that sets this Java System Property for different test cases.

Reply:
Then tests need to be adjusted.  We can't have the main code adjusted because of tests...

Reply:
I would not like to modify MavenCliTest, that's all. And it achieves it's fixtures by setting this system property... and nothing else.

Reply:
> Then tests need to be adjusted. We can't have the main code adjusted because of tests...

Whenever test use `mavenMultiModuleDirectory`, they should be changed to use `cliRequest.workingDirectory`...

Reply:
In all cases, this PR is wrong.  `mavenMultiModuleDirectory` is similar to `session.rootDirectory`, but certainly not to `topDirectory`. 
Just try the following:
```
➜  maven git:(docgen) ✗ mvn help:evaluate -Dexpression=session.topDirectory -N -f maven-core -q -DforceStdout         
<path>/Users/gnodet/work/git/maven/maven-core</path>%                                                                                                                                    ➜  maven git:(docgen) ✗ mvn help:evaluate -Dexpression=maven.multiModuleProjectDirectory -N -f maven-core -q -DforceStdout
/Users/gnodet/work/git/maven%                                                                                                                                                            ```

Reply:
The problem is that your definition of `multiModuleProjectDirectory` is wrong in the PR first comment...